allennlp/training/trainer.py

```diff
@@ -1038,6 +1064,7 @@ def from_params(cls,  # type: ignore
         histogram_interval = params.pop_int("histogram_interval", None)
         should_log_parameter_statistics = params.pop_bool("should_log_parameter_statistics", True)
         should_log_learning_rate = params.pop_bool("should_log_learning_rate", False)
+        log_batch_size_period = params.pop_int("log_batch_size_period", None)
```

Comment:
Do we really need this new parameter? Can't we just log the batch size every `summary_interval` steps? It's not expensive because it's a scalar, and it's not too visually obnoxious because tensorboard compresses graphs by default in the UI.